### PR TITLE
Add profiler instrumentation to MRTK's raycast logic

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change extends the input system profiler markers (#7590) to the raycast implementation.

### CODE REVIEW

File: Assets/MRTK/Services/InputSystem/FocusProvider.cs [hunk 1]
```diff
@@ -887,16 +887,20 @@ public void UnsubscribeFromPrimaryPointerChanged(PrimaryPointerChangedHandler ha
         /// <returns>Pointer Data if the pointing source is registered.</returns>
         private bool TryGetPointerData(IMixedRealityPointer pointer, out PointerData data)
         {
+            Profiler.BeginSample("[MRTK] FocusProvider.TryGetPointerData");
```
Comment (keveleigh) [inline Assets/MRTK/Services/InputSystem/FocusProvider.cs]: How noisy do markers like this end up being? There are some methods in here that I'm not sure how useful profiling them would be. I feel like putting in too many markers could blow up the profiler's list and become harder to be useful.

Comment (davidkline-ms) [reply Assets/MRTK/Services/InputSystem/FocusProvider.cs]: not sure. good question

Comment (davidkline-ms) [reply Assets/MRTK/Services/InputSystem/FocusProvider.cs]: let's back them off until we know they tell us somethin

**Pull request merged**